buffer.c

```diff
@@ -124,63 +61,41 @@ zend_class_entry *double_array_ce;
 zend_object_handlers array_buffer_handlers;
 zend_object_handlers array_buffer_view_handlers;
 
-static long get_long_from_zval(zval *zv)
+static void array_buffer_free(zend_object *object)
 {
-	if (Z_TYPE_P(zv)) {
-		return Z_LVAL_P(zv);
-	} else {
-		long lval;
-		Z_ADDREF_P(zv);
-		convert_to_long_ex(&zv);
-		lval = Z_LVAL_P(zv);
-		zval_ptr_dtor(&zv);
-		return lval;
-	}
-}
-
-static void array_buffer_free_object_storage(buffer_object *intern TSRMLS_DC)
-{
-	zend_object_std_dtor(&intern->std TSRMLS_CC);
+	buffer_object * intern = php_buffer_fetch_object(object);
 
 	if (intern->buffer) {
 		efree(intern->buffer);
 	}
 
+	zend_object_std_dtor(&intern->std);
 	efree(intern);
```

Comment:
Should be dropped, this will cause a double free.

Reply:
Sorry about that. Too bad it wasn't caught from the texts.